executor/index_lookup_hash_join.go

```diff
@@ -433,6 +433,16 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 			joinResult.err = err
 			break
 		}
+		if task.keepOuterOrder {
+			// We need to get a new result holder here because the old
+			// `joinResult` hash been sent to the `resultCh` or to the
+			// `joinChkResourceCh`.
+			joinResult, ok = iw.getNewJoinResult(ctx)
+			if !ok {
+				cancelFunc()
+				return
+			}
+		}
```

Comment:
Is it better to put it in `doJoinInOrder` or `handleTask`?

Reply:
How about putting it [here](https://github.com/pingcap/tidb/blob/master/executor/index_lookup_hash_join.go#L655), the first `defer` function in `doJoinInOrder`?